src/Form/SettingsForm.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Drupal\y_lb\Form;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Settings Form for Small Y.
+ */
+class SettingsForm extends ConfigFormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'ws_admin_settings';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return [
+      'y_lb.admin.settings',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->config('y_lb.admin.settings');
+
+    $form['social_links'] = [
+      '#type' => 'details',
+      '#title' => $this->t('Social Links'),
+      '#open' => TRUE,
+        '#description' => $this->t('In this section, please add links to your pages on the social networks.<br>
+          You can leave the field empty if you don\'t have a page on some network.'),
+    ];
+
+    $form['social_links']['facebook'] = [
+      '#type' => 'url',
+      '#title' => $this->t('Facebook'),
+      '#default_value' => $config->get('social_links.facebook') ?? NULL,
+      '#pattern' => '^(https?:\/\/)?(?:www\.)?facebook\.com(.+)?',
+      '#description' => $this->t('Please add a link to your Facebook page.'),
+    ];
+
+    $form['social_links']['twitter'] = [
+      '#type' => 'url',
+      '#title' => $this->t('X(Twitter)'),
+      '#default_value' => $config->get('social_links.twitter') ?? NULL,
+      '#pattern' => '^(https?:\/\/)?(?:www\.)?(x|twitter)\.com(.+)?',
+      '#description' => $this->t('Please add a link to your X(Twitter) page.'),
+    ];
+
+    $form['social_links']['youtube'] = [
+      '#type' => 'url',
+      '#title' => $this->t('Youtube'),
+      '#default_value' => $config->get('social_links.youtube') ?? NULL,
+      '#pattern' => '^(https?:\/\/)?(?:www\.)?youtube\.com(.+)?',
+      '#description' => $this->t('Please add a link to your Youtube page.'),
+    ];
+
+    $form['social_links']['instagram'] = [
+      '#type' => 'url',
+      '#title' => $this->t('Instagram'),
+      '#default_value' => $config->get('social_links.instagram') ?? NULL,
+      '#pattern' => '^(https?:\/\/)?(?:www\.)?instagram\.com(.+)?',
+      '#description' => $this->t('Please add a link to your Instagram page.'),
+    ];
+
+    $form['social_links']['linkedin'] = [
+      '#type' => 'url',
+      '#title' => $this->t('Linkedin'),
+      '#default_value' => $config->get('social_links.linkedin') ?? NULL,
+      '#pattern' => '^(https?:\/\/)?(?:www\.)?linkedin\.com(.+)?',
+      '#description' => $this->t('Please add a link to your LinkedIn page.'),
+    ];
+
+    return parent::buildForm($form, $form_state);
+  }
```

Comment:
This form can be optimized.
Something like:  
```
public function buildForm(array $form, FormStateInterface $form_state) {
    $config = $this->config('y_lb.admin.settings');
    $form['#description'] = $this->t('Configure social media links for your site.');

    $social_networks = [
      'facebook' => 'Facebook',
      'twitter' => 'X (Twitter)',
      'youtube' => 'YouTube',
      'instagram' => 'Instagram',
      'linkedin' => 'LinkedIn',
    ];

    $form['social_links'] = [
      '#type' => 'details',
      '#title' => $this->t('Social Links'),
      '#open' => TRUE,
      '#description' => $this->t('Please add links to your social network pages. Leave fields empty if they are not applicable.'),
    ];

    foreach ($social_networks as $network => $label) {
      $form['social_links'][$network] = [
        '#type' => 'url',
        '#title' => $this->t($label),
        '#default_value' => $config->get("social_links.$network"),
        '#description' => $this->t("Please add a link to your @label page.", ['@label' => $label]),
      ];
    }

    return parent::buildForm($form, $form_state);
  }
```

I removed the `#pattern` because the `url` field type should already provide basic validation, if it doesn't then we can expand the `$social_networks` array with this data.

Reply:
@rollki, thank you for your comment. 
I've added the `#pattern` validation to ensure the editor has put the right URL for a particular social network.
 I can implement your vision, but I prefer using as simple code as possible. I don't understand why we need to make the code complicated when it can be simple.

Reply:
@rollki 
I understand your idea about decreasing code duplication. But in this case, I think it's optimization for optimization. But I could be wrong.
`efforts that may be needed in the future for extending (eg adding a new social network).`
I don't think that the difference will be that much in this particular case